### PR TITLE
Image's placeholder for new document

### PR DESCRIPTION
**Pull request by drucelweisse**
# Description
This pr adds button on EditorView to open image file and save it to document. Add menu item to open file

## Changes:
- added NSButton to Editor storyboard
- added EditorViewController addImageButtonTapped() function in EditorViewController, hooked with new button
- added EditorViewController addMenuItem() to create menu item, hooked width addImageButtonTapped()

## Todo:
- [X] Create button "add image" that allows to select from Finder
- [x] Add menu item for image attachment

## Screenshots
Screenshot 2022-09-07 at 22 58 10

### CODE REVIEW

**Comment by drucelweisse** [comment]
Added menu item interaction video sample

https://user-images.githubusercontent.com/36012972/189497408-ab85c547-6e86-4714-97b1-738d44242609.mov

**Comment by drucelweisse** [comment]
Which file types should we support for image? Currently, panel will try to open any provided file.

File: VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift [hunk 1]
```diff
@@ -129,8 +144,28 @@ public class EditorViewController: NSViewController {
...
+    @objc func addImageButtonTapped() {
+        guard let window = view.window else { return }
+        let imagePanel = NSOpenPanel()
+        imagePanel.canChooseFiles = true
+        imagePanel.canChooseDirectories = false
+        imagePanel.allowsMultipleSelection = false
+        imagePanel.beginSheetModal(for: window) { [weak self] response in
+            if response == .OK {
+                if let url = imagePanel.url, let image = NSImage(contentsOf: url) {
+                    self?.presenter.update(image: image)
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift:157]: Can you extract openPanel-related and image setting code to separate functions? At this function we will have high-level code: request image and present image.

File: VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift [hunk 2]
```diff
@@ -40,14 +40,29 @@ public class EditorViewController: NSViewController {
...
         DispatchQueue.main.async {
             self.presenter.didLoad(
                 ui: self.view().controlsView)
             self.setImage()
             self.addMouseTracking()
+            self.addMenuItem()
         }
     }
     
+    // TODO: try to extract?
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift:55]: Support for menu is cool? Definitely I don't know where to place it :D 

This place is fine for know

**Review by akaDuality — CHANGES_REQUESTED** [review]

File: VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift (hunk 1 above)
Comment (drucelweisse) [reply VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift]: resolved in https://github.com/VODGroup/VoiceOverDesigner/pull/112/commits/92c28a0a6c7a18946663993a7f4b75a3efffd46f

File: VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift [hunk 3]
```diff
@@ -129,8 +144,40 @@ public class EditorViewController: NSViewController {
     public func delete(model: A11yDescription) {
         presenter.delete(model: model)
     }
+    
+    @objc func addImageButtonTapped() {
+        Task {
+            if let image = await requestImage() {
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Sources/Editor/EditorViewController.swift:150]: Async syntax is awesome!

**Review by akaDuality — APPROVED** [review]

**Pull request merged**